src/cbc_sdk/platform/grants.py

```diff
@@ -280,22 +396,30 @@ def _update_object(self):
         save_info = copy.deepcopy(self._info)
         if 'principal' not in save_info:
             raise ApiError("principal must be specified for a Grant object")
-        save_info['profiles'] = [copy.deepcopy(prof._info) for prof in self._profiles if 'profile_uuid' in prof._info]
+        is_update = all([key in save_info for key in ['created_by', 'updated_by', 'create_time', 'update_time']])
 
-        if all([key in save_info for key in ['created_by', 'updated_by', 'create_time', 'update_time']]):
+        if is_update:
             log.debug("Updating the Grant object for principal {0:s}".format(save_info['principal']))
+            save_info['profiles'] = [copy.deepcopy(prof._info)
+                                     for prof in self._profiles if 'profile_uuid' in prof._info]
             url = self.urlobject_single.format(self._cb.credentials.org_key, save_info['principal'])
             ret = self._cb.put_object(url, save_info)
         else:
             log.debug("Creating a new Grant object for principal {0:s}".format(save_info['principal']))
+            save_info['profiles'] = []
+            for profile in self._profiles:
+                # presumed to be all new profiles, so create UUIDs for them
+                profile._info['profile_uuid'] = str(uuid.uuid4())
```

Comment:
Have you confirmed if this is necessary to create your own uuid? On the creation of a grant will it accept profiles without uuids?